src/app/ReadClient.h

```diff
@@ -83,6 +83,15 @@ class ReadClient : public Messaging::ExchangeDelegate
 
         virtual ~Callback() = default;
 
+        /**
+         * Used to notify a (maybe empty) report data is received from peer and the subscription and the peer is alive.
+         *
+         * This object MUST continue to exist after this call is completed. The application shall wait until it
+         * receives an OnDone call to destroy the object.
+         *
+         */
+        virtual void RefreshLiveness(const ReadClient & apReadClient) {}
```

Comment:
Naming-wise, this would have been better as "NotifySubscriptionActive".  That would make it clear this is only relevant to subscriptions and would make it much clearer just from the function name what the notification is about.

Reply:
https://github.com/project-chip/connectedhomeip/pull/31354